sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuRoundRobinPartitioning.scala

```diff
@@ -42,21 +43,28 @@ case class GpuRoundRobinPartitioning(numPartitions: Int)
 
   def partitionInternal(batch: ColumnarBatch): (Array[Int], Array[GpuColumnVector]) = {
     val sparkTypes = GpuColumnVector.extractTypes(batch)
-    withResource(GpuColumnVector.from(batch)) { table =>
-      if (numPartitions == 1) {
-        val columns = (0 until table.getNumberOfColumns).zip(sparkTypes).map {
-          case(idx, sparkType) =>
-            GpuColumnVector.from(table.getColumn(idx).incRefCount(), sparkType)
-        }.toArray
-        return (Array(0), columns)
-      }
-      withResource(table.roundRobinPartition(numPartitions, getStartPartition)) { partedTable =>
-        val parts = partedTable.getPartitions
-        val columns = (0 until partedTable.getNumberOfColumns.toInt).zip(sparkTypes).map {
-          case(idx, sparkType) =>
-            GpuColumnVector.from(partedTable.getColumn(idx).incRefCount(), sparkType)
-        }.toArray
-        (parts, columns)
+    // Increase ref count since the caller will close the batch also.
+    val spillableBatch = SpillableColumnarBatch(GpuColumnVector.incRefCounts(batch),
+      SpillPriorities.ACTIVE_ON_DECK_PRIORITY)
+    withRetryNoSplit(spillableBatch) { sb =>
+      withResource(GpuColumnVector.from(sb.getColumnarBatch())) { table =>
```

Comment:
We need to close the batch separate from the table. I think there is a leak here.